### ISSUE TITLE
Roles middleware can't use same URL for different roles.

### ISSUE THREAD

**Issue by TobbyReign** [spatie/laravel-permission#2337]
**Describe the bug**
User aren't redirected to their assigned view based on their role if url are the same (/dashboard).

**Versions**
- spatie/laravel-permission package version: 5.9.1
- Laravel 10.0.3

PHP version: 8.2.3

**To Reproduce**
Here is my web.php
Only url is the same, view and route name isn't the same.

```
Route::group(['middleware' => ['role:admin']], function () {
    Route::get('/dashboard', function () {
        return view('admin.dashboard');
    })->middleware(['auth', 'verified'])->name('admin.dashboard');
    
});


Route::group(['middleware' => ['role:user']], function () {
    Route::get('/dashboard', function () {
        return view('user.dashboard');
    })->middleware(['auth', 'verified'])->name('user.dashboard');
    
});
```

It look like it only lookup for the last line of code for roles in the web.php.
Basically, user role can view his own view but admin get ' User does not have the right roles. '

**Expected behavior**
  Based on authentified user role, redirect to their respective assigned view while they hit /dashboard

**Environment:**
 - Ubuntu 22.04 (Apache2)
 - Laravel built-in server (artisan serve)

**Comment by parallels999** [comment]
I don't see much of it, as far as I understood, Laravel cannot have two equal routes, that is, if you put two equal ones, it will only take the last

**Comment by drbyte** [comment]
@parallels999 is correct.

You've made 2 definitions for the same `get('/dashboard')` route. So only the last one "works".
Either change the `get` URL so you can have 2 different routes (to coincide with your route 'names'), or combine the logic into one route (and one route name).

**Comment by parallels999** [comment]
@drbyte still, it's not a bug, maybe must be on discussions section
